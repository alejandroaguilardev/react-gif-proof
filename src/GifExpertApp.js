import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = []}) => {
    const [categories, setCategories] = useState( defaultCategories );
    
    // const handleAdd = () => {
    //     setCategories(cats => [...cats, 'HunterXHunter']);
    //     // setCategories([...categories, 'HunterXHunter']);
    //     //setCategories(categories => categories.concat('HunterXHunter')) // Opci√≥n 1
    //     //setCategories(categories.concat('HunterXHunter')); // Opcion 2 
    // }
  return <>
    <h2>GifExpertApp</h2>
    <AddCategory  setCategories = { setCategories }  />
    <hr/>

    <ul>
     { 
        categories.map( category =>  (
            <GifGrid 
              key = { category }
              category = { category } 
            />
        ))
     }
    </ul>
  </>;
};

export default GifExpertApp;
