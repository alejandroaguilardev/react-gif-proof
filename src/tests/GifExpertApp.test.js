import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('Prueba gifExper', ()=>{
    let wrapper = shallow( <GifExpertApp />);

    test('Llama al componente', ()=>{
       expect( wrapper ).toMatchSnapshot();
    });

    test('debe dem ostrar una lista de categorias', ()=>{
        const categorias = ['One punch', 'dragon'];
        wrapper = shallow( <GifExpertApp defaultCategories={categorias}/>);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length).toBe(categorias.length);

        
     })
})