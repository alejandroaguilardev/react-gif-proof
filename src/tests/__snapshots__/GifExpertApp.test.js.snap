// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba gifExper Llama al componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Prueba gifExper debe dem ostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="dragon"
      key="dragon"
    />
  </ul>
</Fragment>
`;
